scheduler/ipp.c

```diff
@@ -2449,18 +2449,17 @@ add_printer(cupsd_client_t  *con,	/* I - Client connection */
       for (i = 0; i < supported->num_values; i ++)
         if (!strcmp(supported->values[i].string.text,
                     attr->values[0].string.text))
-          break;
+          goto port_monitor_is_ok;
     }
 
-    if (!supported || i >= supported->num_values)
-    {
-      send_ipp_status(con, IPP_NOT_POSSIBLE, _("Bad port-monitor \"%s\"."),
-        	      attr->values[0].string.text);
-      if (!modify)
-        cupsdDeletePrinter(printer, 0);
+    send_ipp_status(con, IPP_NOT_POSSIBLE, _("Bad port-monitor \"%s\"."),
+                    attr->values[0].string.text);
+    if (!modify)
+      cupsdDeletePrinter(printer, 0);
 
-      return;
-    }
+    return;
+
```

Comment:
Please don't introduce goto's for this. The only time we use goto is for a simple form of exception handling at the bottom of a function.